src/inmanta/deploy/scheduler.py

```diff
@@ -386,7 +387,15 @@ async def report_resource_state(
                     if event_listeners:
                         # do not pass deploying tasks because for event propagation we really want to start a new one,
                         # even if the current intent is already being deployed
-                        self._work.deploy_with_context(event_listeners, deploying=set())
+                        priority = self._work.agent_queues.in_progress.get(Deploy(resource=resource))
+                        if priority is None:
+                            LOGGER.warning(
+                                "Deploy task for resource %s not found on in_progress."
+                                "Assigning a priority of NEW_VERSION_DEPLOY to the events it produced",
+                                resource,
+                            )
+                            priority = TaskPriority.NEW_VERSION_DEPLOY
+                        self._work.deploy_with_context(event_listeners, priority=priority, deploying=set())
```

Comment:
@wouterdb @sanderr This is the part that we discussed. 
If we don't find the task in the queue I opted for logging a warning and assigning a "default" priority. What do you think?